File: algorithms/sort/selection_sort.py
def selection_sort(arr):
    # Iterate through the array
    for i in range(len(arr)): #
        # Assume the current position is the min value
        min_index = i
        # Iterate through the remaining unsorted array values
        for j in range(i+1, len(arr)):
            # If we find a lower value
            if arr[j] < arr[min_index]: # If we find a lower value
                # Then update our min value index
                min_index = j
        # Swap the value in the next lowest index with the actual lowest value
        arr[i], arr[min_index] = arr[min_index], arr[i]

    return arr

File: algorithms/sort/selection_sort.rb

def selection_sort(arr)
  0.upto(arr.size - 1) do |smallest_index|
    start = smallest_index + 1

    start.upto(arr.size - 1) do |index|
      if arr[index] <= arr[smallest_index]
        arr[index], arr[smallest_index] = arr[smallest_index], arr[index]
      end
    end
  end

  arr
end
